### PR TITLE
More taps

### PR DESCRIPTION
**Pull request by albertpastrana**
This PR adds some useful functions we have been using as implicits in other projects.

@nathankleyn it would be great if you could take a look at it

### CODE REVIEW

File: result-async/src/test/scala/uscala/concurrent/result/AsyncResultSpec.scala [hunk 1]
```diff
@@ -137,6 +137,64 @@ class AsyncResultSpec(implicit ee: ExecutionEnv) extends Specification with Scal
     }
   }
 
+  "tap" >> {
```
Comment (nathankleyn) [inline result-async/src/test/scala/uscala/concurrent/result/AsyncResultSpec.scala:140]: Looks like we're missing tests for `tapOk` - even though it's an alias, we should probably test it anyway (for coverage and catching accidents).

File: result/src/test/scala/uscala/result/ResultSpec.scala [hunk 2]
```diff
@@ -123,6 +123,27 @@ class ResultSpec extends Specification with ScalaCheck {
     }
   }
 
+  "tapFail" >> {
```
Comment (nathankleyn) [inline result/src/test/scala/uscala/result/ResultSpec.scala]: Again, probably best to test `tapOk` even if it is just an alias.

File: result/src/test/scala/uscala/result/ResultSpec.scala [hunk 3]
```diff
@@ -285,6 +306,35 @@ class ResultSpec extends Specification with ScalaCheck {
...
+      "should create a Fail if it's Failure" >> prop { e: Exception =>
+        Failure(e).toResult must_=== Fail(e)
+      }
+      "should create an Ok if it's Success" >> prop { n: Int =>
+        Success(n).toResult must_=== Ok(n)
+      }
+    }
+  }
+
   "sequence for options" >> {
```
Comment (nathankleyn) [inline result/src/test/scala/uscala/result/ResultSpec.scala:350]: I think you're missing a spec for `fromSomething`?

**Review by nathankleyn — CHANGES_REQUESTED** [review]
Couple of missing specs, but implementation looks great and these are awesome additions! 👍

File: result/src/test/scala/uscala/result/ResultSpec.scala (hunk 3 above)
Comment (albertpastrana) [reply result/src/test/scala/uscala/result/ResultSpec.scala:350]: sorry, I pushed this one by mistake, I'll remove the function

**Review by nathankleyn — APPROVED** [review]
LGTM 👍

**Pull request merged**